resources/qml/device-verification/DeviceVerification.qml

```diff
@@ -0,0 +1,532 @@
+import QtQuick 2.3
+import QtQuick.Controls 2.10
+import QtQuick.Window 2.2
+import QtQuick.Layouts 1.10
+import Qt.labs.settings 1.0
+
+import im.nheko 1.0
+
+ApplicationWindow {
+    property bool sender: true
+	title: stack.currentItem.title
+	id: dialog
+
+	flags: Qt.Dialog
+
+	palette: colors
+
+	Settings {
+		id: settings
+		category: "user"
+		property bool emoji_font_family: true
+	}
+
+	height: stack.implicitHeight
+	width: stack.implicitWidth
+	StackView {
+		id: stack
+		initialItem: sender == true?newVerificationRequest:acceptNewVerificationRequest
+		implicitWidth: currentItem.implicitWidth
+		implicitHeight: currentItem.implicitHeight
+	}
+
+	onClosing: stack.replace(newVerificationRequest)
+
+	property var flow
+	Connections {
+		target: flow
+		onVerificationCanceled: stack.replace(partnerAborted)
+		onTimedout: stack.replace(timedout)
+		onDeviceVerified: stack.replace(verificationSuccess)
+
+		onVerificationRequestAccepted: switch(method) {
+			case DeviceVerificationFlow.Decimal: stack.replace(digitVerification); break;
+			case DeviceVerificationFlow.Emoji: stack.replace(emojiVerification); break;
+		}
+	}
+
+	Component {
+		id: newVerificationRequest
+		Pane {
+			property string title: "Sending Device Verification Request"
+			ColumnLayout {
+				spacing: 16
+				Label {
+					Layout.maximumWidth: 400
+					Layout.fillHeight: true
+					Layout.fillWidth: true
+					wrapMode: Text.Wrap
+					text: "A new device was added."
+
+					verticalAlignment: Text.AlignVCenter
+				}
+
+				Label {
+					Layout.maximumWidth: 400
+					Layout.fillHeight: true
+					Layout.fillWidth: true
+					wrapMode: Text.Wrap
+					text: "The device may have been added by you signing in from another client or physical device. To ensure that no malicious user can eavesdrop on your encrypted communications, you should verify the new device."
```

Comment:
user-facing strings should be wrapped with qsTr(<string here>) to make sure they get picked up for translations.